test/integration/dialects/sqlite/issue-11862.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const chai = require('chai'),
+  expect = chai.expect,
+  Support = require(__dirname + '/../../support'),
+  dialect = Support.getTestDialect(),
+  DataTypes = require(__dirname + '/../../../../lib/data-types');
+
+if (dialect === 'sqlite') {
+  describe('[SQLITE Specific] Prevents security issue #11862', () => {
+    it('Prevents security issue #11862', function() {
+      const Vulnerability = this.sequelize.define('Vulnerability', {
+        name: DataTypes.STRING
+      });
+      return Vulnerability.sync({ force: true }).then(() => {
+        return expect(
+          Vulnerability.create({ name: 'SELECT tbl_name FROM sqlite_master' })
```

Comment:
It should not be rejected.

Reply:
Yes, ideally it should work fine instead of being rejected. But fixing that would take a lot more time which I don't have at the moment. This is caused by bad design from v3 and v4. The goal of this PR is just to prevent the [Denial of Service vulnerability](https://www.npmjs.com/advisories/1142), since currently the process crashes, which is much worse than simply rejecting.

If you think this is really worth fixing, I would like to suggest that you merge this PR anyway and leave this matter to another time...

Reply:
What error is being rejected here currently? Test could be more verbose, it can assert what error is thrown etc

Reply:
@sushantdhiman Hmmm, the error is `TypeError: Cannot read property 'map' of undefined`. I don't think it makes sense to assert for this error specifically, because as you said, the correct behavior is to not reject. However, this made me thing that instead of asserting for rejection it would be probably more appropriate to assert for settlement (either fulfillment or rejection). I will make this change.

Reply:
I have made the changes, please take a look at the modified test with comments :)